metadata-ingestion/src/datahub/sdk/lineage_client.py

```diff
@@ -227,9 +240,140 @@ def add_dataset_transform_lineage(
+                    ]
+
+                    if upstream_cols:
+                        column_mapping[col_lineage.downstream.column] = upstream_cols
+
+            # Add lineage, including query text only for the first upstream
+            self.add_dataset_transform_lineage(
+                upstream=upstream_table,
+                downstream=downstream_urn,
+                column_lineage=column_mapping or None,
+                query_text=query_text if i == 0 else None,
+            )
+
+    def add_datajob_lineage(
+        self,
+        *,
+        datajob: DatajobUrnOrStr,
+        upstreams: Optional[List[Union[DatasetUrnOrStr, DatajobUrnOrStr]]] = None,
+        downstreams: Optional[List[DatasetUrnOrStr]] = None,
+    ) -> None:
+        """
+        Add lineage between a datajob and datasets/datajobs.
+
+        Args:
+            datajob: The datajob URN to connect lineage with
+            upstreams: List of upstream datasets or datajobs that serve as inputs to the datajob
+            downstreams: List of downstream datasets that are outputs of the datajob
+        """
+
+        datajob_urn = str(datajob)
+        if not datajob_urn.startswith("urn:li:dataJob:"):
+            raise SdkUsageError(
+                f"The datajob parameter must be a DataJob URN, got {datajob_urn}"
+            )
+
+        # Initialize the patch builder for the datajob
+        patch_builder = DataJobPatchBuilder(datajob_urn)
+
+        # Process upstream connections (inputs to the datajob)
+        if upstreams:
+            for upstream in upstreams:
+                upstream_urn = str(upstream)
+
+                if upstream_urn.startswith("urn:li:dataset:"):
+                    # Add dataset as input to the datajob
+                    patch_builder.add_input_dataset(upstream_urn)
+                elif upstream_urn.startswith("urn:li:dataJob:"):
+                    # Add datajob as input to the datajob
+                    patch_builder.add_input_datajob(upstream_urn)
+                else:
+                    raise SdkUsageError(
+                        f"Upstream URN {upstream_urn} must be either a dataset or datajob URN"
+                    )
+
+        # Process downstream connections (outputs from the datajob)
+        if downstreams:
+            for downstream in downstreams:
+                downstream_urn = str(downstream)
+
+                if downstream_urn.startswith("urn:li:dataset:"):
+                    # Add dataset as output from the datajob
+                    patch_builder.add_output_dataset(downstream_urn)
+                else:
+                    raise SdkUsageError(
+                        f"Downstream URN {downstream_urn} must be a dataset URN"
+                    )
+
+        # Apply the changes to the entity
+        if upstreams or downstreams:  # Only update if there are actual changes
+            self._client.entities.update(patch_builder)
```

Comment:
maybe throw an error if nothing was passed in?

Reply:
done